vendor/bat-native-ads/src/bat/ads/internal/guid_util.cc

```diff
@@ -0,0 +1,27 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/guid_util.h"
+
+#include "base/guid.h"
+#include "third_party/abseil-cpp/absl/types/optional.h"
+
+namespace {
+
+absl::optional<std::string> g_guid;
```

Comment:
global with with non-trivial destructor is banned in google style guide

Reply:
Fixed